modules/_labs/side-panel/react/lib/SidePanel.tsx

```diff
@@ -0,0 +1,243 @@
+
+  React.useEffect(() => {
+    if (typeof onStateChange !== 'undefined') {
+      onStateChange(internalState);
+    }
+  }, [internalState, onStateChange]);
+
+  const motion = {
+    collapse: createKeyframes(collapsedWidth, width),
+    expand: createKeyframes(width, collapsedWidth),
+  };
+
+  const handleClick = () => {
+    setCollapsed(!collapsed);
+  };
+
+  const handleAnimationEnd = (event: React.AnimationEvent<HTMLDivElement>) => {
+    if (event.currentTarget === event.target) {
+      if (
+        event.animationName === motion.collapse.name ||
+        event.animationName === motion.expand.name
+      ) {
+        setInternalState(collapsed ? 'collapsed' : 'expanded');
+      }
+    }
+
+    if (typeof onAnimationEnd !== 'undefined') {
+      onAnimationEnd(event);
+    }
+  };
+
+  const handleAnimationStart = (event: React.AnimationEvent<HTMLDivElement>) => {
+    if (event.currentTarget === event.target) {
+      if (
+        event.animationName === motion.collapse.name ||
+        event.animationName === motion.expand.name
+      ) {
+        setInternalState(collapsed ? 'collapsing' : 'expanding');
+      }
+    }
+
+    if (typeof onAnimationStart !== 'undefined') {
+      onAnimationStart(event);
+    }
+  };
+
+  // Note: Depending on the collapsed width, the button could "jump" to it's final position.
+  const buttonStyle = css({
+    position: 'absolute',
+    top: spacing.l,
+    right: internalState === 'collapsed' ? 0 : origin === 'left' ? spacing.s : undefined,
+    left: internalState === 'collapsed' ? 0 : origin === 'right' ? spacing.s : undefined,
+    margin: internalState === 'collapsed' ? 'auto' : undefined,
+    transform: collapsed
+      ? `scaleX(${origin === 'left' ? '1' : '-1'})`
+      : `scaleX(${origin === 'left' ? '-1' : '1'})`,
+  });
+
+  // TODO: if we're in controlled mode should we ship a hook that spreads aria-controls and aria-expanded
+  return (
+    <div
+      id={isControlled ? id : sidePanelId}
+      role="region"
+      css={[
+        {
+          overflow: 'hidden',
+          position: 'relative',
+          boxSizing: 'border-box',
+          width: collapsed ? collapsedWidth : width,
+          maxWidth: collapsed ? collapsedWidth : width,
+          minWidth: collapsed ? collapsedWidth : width,
+          height: height,
+          minHeight: height,
+          maxHeight: height,
+          // mounted.current will be false on the first render, thus you won't get an unwanted animation here
+          // Will animate again if you force a re-render (like in Storybook)
+          animation: mounted.current
+            ? `${collapsed ? motion.expand : motion.collapse} 200ms ease-out`
+            : undefined,
+        },
+        containerVariantStyle[variant],
+      ]}
+      onAnimationEnd={handleAnimationEnd}
+      onAnimationStart={handleAnimationStart}
+      {...elemProps}
+    >
+      {typeof collapsedProp === 'undefined' ? (
+        <IconButton
```

Comment:
I was back and forth on this one. But eventually I just figured that this was a good enough first pass and we should get this out there and see what people think.

Here's how it played out in my head. The `IconButton` look (icon, variant, and position) is something we aim to specify and/or standardize as a design system so that was a fixed option. However, a user can most certainly use a custom `IconButton` as one of the children when it's a controlled component.

In other words, because Side Panel consistency has been such a hot topic internally, I wanted to make the standard choice easy, and the non-standard choice doable with ease being less of a factor.